src/TropicalGeometry/linear_space.jl

```diff
@@ -335,13 +349,72 @@ end
+function ensure_pluecker_data!(TropL::TropicalLinearSpace)
+
+    if has_attribute(TropL,:tropical_pluecker_vector)
+        return
+    end
+
+    # check if tropical semiring map is available
+    if has_attribute(TropL,:tropical_semiring_map)
+        # if yes, we compute the tropical pluecker vector from the algebraic data
+        nu = tropical_semiring_map(TropL)
+
+        # check if algebraic pluecker vector is available
+        # if yes, compute the tropical pluecker vector
+        if has_attribute(TropL,:algebraic_pluecker_vector)
+            algebraicPlueckerVector = get_attribute(TropL,:algebraic_pluecker_vector)
+            tropicalPlueckerVector = nu.(algebraicPlueckerVector)
+            set_attribute!(TropL,:tropical_pluecker_vector,tropicalPlueckerVector)
+            return
+        end
+        # check if algebraic matrix is available
+        # if yes, compute the algebraic pluecker vector and rerun
+        if has_attribute(TropL,:algebraic_matrix)
+            A = get_attribute(TropL,:algebraic_matrix)
+            n = ncols(A)
+            k = rank(A)
+            plueckerIndices = AbstractAlgebra.combinations(1:n,k)
+            algebraicPlueckerVector = AbstractAlgebra.minors(A,k)
+            set_attribute!(TropL,:pluecker_indices,plueckerIndices)
+            set_attribute!(TropL,:algebraic_pluecker_vector,algebraicPlueckerVector)
+            return ensure_pluecker_data!(TropL)
+        end
+        # check if algebraic ideal is available
+        # if yes, compute the algebraic matrix and rerun
+        if has_attribute(TropL,:algebraic_ideal)
+            I = get_attribute(TropL,:algebraic_ideal)
+            A = matrix([[coeff(g,xi) for xi in gens(base_ring(I))] for g in gens(I)])
+            set_attribute!(TropL,:algebraic_matrix,A)
+            return ensure_pluecker_data!(TropL)
+        end
+    else
+        # if not, we have to compute the tropical pluecker vector from the tropical data
+        if has_attribute(TropL,:tropical_matrix)
+            A = get_attribute(TropL,:tropical_matrix)
+            n = ncols(A)
+            k = rank(A)
+            plueckerIndices = AbstractAlgebra.combinations(1:n,k)
+            tropicalPlueckerVector = AbstractAlgebra.minors(A, k)
+            println("tropical matrix -> tropical pluecker")
+            set_attribute!(TropL,:pluecker_indices,plueckerIndices)
+            set_attribute!(TropL,:tropical_pluecker_vector,tropicalPlueckerVector)
+            return
+        end
+    end
+
+    @req false "insufficient data to compute tropical pluecker vector"
+
+end
+
 @doc raw"""
     pluecker_indices(TropL::TropicalLinearSpace)
 
 Return the Pluecker indices used to construct `TropL`.  Raises an error, if it is not cached.
 """
 function pluecker_indices(TropL::TropicalLinearSpace)
-    @req has_attribute(TropL,:pluecker_indices) "no pluecker indices cached"
+    ensure_pluecker_data!(TropL)
     return get_attribute(TropL,:pluecker_indices)
```

Comment:
But `ensure_pluecker_data!` does not always set `pluecker_indices`, so shouldn't the `@req` stay, or a similar new test be added?


Looking at the code, how about something like this instead (ideally with a more specific return type for `algebraic_matrix`, but since you effectively use `Any` right now, I put that there):
```julia
@attr Vector{Vector{In64}} function pluecker_indices(TropL::TropicalLinearSpace)
    if has_attribute(TropL,:tropical_matrix)
        A = get_attribute(TropL,:tropical_matrix)
    else
        A = algebraic_matrix(TropL)
    end
    n = ncols(A)
    k = rank(A)
    return AbstractAlgebra.combinations(1:n,k)
end

@attr Any function algebraic_matrix(TropL::TropicalLinearSpace)
    I = get_attribute(TropL,:algebraic_ideal)
    @req I !== nothing "insufficient data to compute algebraic matrix"
    A = matrix([[coeff(g,xi) for xi in gens(base_ring(I))] for g in gens(I)])
    return A
end
```

I think you can rewrite the other function in a similar vein -- and then even dealing with the case I asked about elsewhere ("what if `tropical_matrix ` and `tropical_semiring_map` are known but none of `algebraic_pluecker_vector`, `algebraic_matrix`, `algebraic_ideal`?") becomes relatively simple.

Reply:
Ah, you got me there.  That is a very good point, I will take a look at it later!

Reply:
@fingolfin I like your suggestion, I'll make the necessary changes now!

Reply:
One question: what is the `@attr` for in `@attr Any function algebraic_matrix(TropL::TropicalLinearSpace)`?

Reply:
`@attr` automatically enables attribute storage for the function. So the result is stored and looked up in the attribute `:algebraic_matrix`. See also the doc string for `@attr`.